include/dgl/aten/array_ops.h

```diff
@@ -24,8 +24,9 @@ namespace aten {
 //////////////////////////////////////////////////////////////////////
 
 /*! \return A special array to represent null. */
-inline NDArray NullArray() {
-  return NDArray::Empty({0}, DLDataType{kDLInt, 64, 1}, DLContext{kDLCPU, 0});
+inline NDArray NullArray(const DLDataType& dtype = DLDataType{kDLInt, 64, 1},
+                         const DLContext& ctx = DLContext{kDLCPU, 0}) {
+  return NDArray::Empty({0}, dtype, ctx);
```

Comment:
I agree it isn't very natural to let NullArray have dtype and context. This is just to avoid breaking some existing context and Idtype checkings that didn't handle null array properly.